python/mirascope/llm/content/image.py

```diff
@@ -57,24 +64,117 @@ def from_url(
+                else:
+                    mime_type = "image/jpeg"
+            else:
+                ext = Path(urlparse(url).path).suffix.lower()
+                mime_type_map: dict[str, ImageMimeType] = {
+                    ".png": "image/png",
+                    ".jpg": "image/jpeg",
+                    ".jpeg": "image/jpeg",
+                    ".webp": "image/webp",
+                    ".gif": "image/gif",
+                    ".heic": "image/heic",
+                    ".heif": "image/heif",
+                }
+                mime_type = mime_type_map.get(ext, "image/jpeg")
+
+            return cls(
+                source=Base64ImageSource(
+                    type="base64_image_source",
+                    data=data,
+                    mime_type=mime_type,
+                )
+            )
+        else:
+            return cls(source=URLImageSource(type="url_image_source", url=url))
 
     @classmethod
     def from_file(
         cls,
         file_path: str,
         *,
-        mime_type: ImageMimeType | None,
+        mime_type: ImageMimeType | None = None,
     ) -> "Image":
         """Create an `Image` from a file path."""
-        raise NotImplementedError
+        path = Path(file_path)
+        if not path.exists():
+            raise FileNotFoundError(f"File not found: {file_path}")
+
+        with open(path, "rb") as f:
+            data = base64.b64encode(f.read()).decode("utf-8")
+
+        final_mime_type: ImageMimeType
+        if mime_type is None:
+            ext = path.suffix.lower()
+            mime_type_map: dict[str, ImageMimeType] = {
+                ".png": "image/png",
+                ".jpg": "image/jpeg",
+                ".jpeg": "image/jpeg",
+                ".webp": "image/webp",
+                ".gif": "image/gif",
+                ".heic": "image/heic",
+                ".heif": "image/heif",
+            }
+            final_mime_type = mime_type_map.get(ext, "image/jpeg")
+        else:
+            final_mime_type = mime_type
+
+        return cls(
+            source=Base64ImageSource(
+                type="base64_image_source",
+                data=data,
+                mime_type=final_mime_type,
+            )
+        )
 
     @classmethod
     def from_bytes(
         cls,
         data: bytes,
         *,
-        mime_type: ImageMimeType | None,
+        mime_type: ImageMimeType | None = None,
     ) -> "Image":
         """Create an `Image` from raw bytes."""
-        raise NotImplementedError
+        encoded_data = base64.b64encode(data).decode("utf-8")
+
+        final_mime_type: ImageMimeType
+        if mime_type is None:
+            # Check magic bytes to determine image format
+            if data.startswith(b"\x89PNG\r\n\x1a\n"):
+                final_mime_type = "image/png"
+            elif data.startswith(b"\xff\xd8\xff"):
+                final_mime_type = "image/jpeg"
+            elif data.startswith(b"RIFF") and data[8:12] == b"WEBP":
```

Comment:
If you provide data that's too small, you'll get an IndexError. You might want to check the data size.

Reply:
Added a check